url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/server/genericServer/LocalStringServerSocket.java

```diff
@@ -0,0 +1,422 @@
+
+    /** Waiting client connections (client-peer sides); Added by connectClient, removed by accept method */
+    protected Vector acceptQueue;
+
+    private String socketName;
+    boolean out_setEOF;
+
+    private Object sync_out_setEOF;  // For synchronized methods, so we don't sync on "this".
+
+    public LocalStringServerSocket(String name)
+    {
+        socketName = name;
+        allConnected = new Vector();
+        acceptQueue = new Vector();
+        out_setEOF = false;
+        sync_out_setEOF = new Object();
+        allSockets.put(name, this);
+    }
+
+    /**
+     * Find and connect to stringport with this name.
+     * Intended to be called by client thread.
+     * Will block-wait until the server calls accept().
+     * Returns a new client connection after accept.
+     *
+     * @param name Stringport server name to connect to
+     *
+     * @throws ConnectException If stringport name is not found, or is EOF,
+     *                          or if its connect/accept queue is full.
+     * @throws IllegalArgumentException If name is null
+     */
+    public static LocalStringConnection connectTo(String name)
+        throws ConnectException, IllegalArgumentException
+    {
+        return connectTo (name, new LocalStringConnection());
+    }
+
+    /**
+     * Find and connect to stringport with this name.
+     * Intended to be called by client thread.
+     * Will block-wait until the server calls accept().
+     *
+     * @param name Stringport server name to connect to
+     * @param client Existing unused connection object to connect with
+     *
+     * @throws ConnectException If stringport name is not found, or is EOF,
+     *                          or if its connect/accept queue is full.
+     * @throws IllegalArgumentException If name is null, client is null,
+     *                          or client is already peered/connected.
+     *
+     * @return client parameter object, connected to a LocalStringServer
+     */
+    public static LocalStringConnection connectTo(String name, LocalStringConnection client)
+        throws ConnectException, IllegalArgumentException
+    {
+        if (name == null)
+            throw new IllegalArgumentException("name null");
+        if (client == null)
+            throw new IllegalArgumentException("client null");
+        if (client.getPeer() != null)
+            throw new IllegalArgumentException("client already peered");
+
+        if (! allSockets.containsKey(name))
+            throw new ConnectException("LocalStringServerSocket name not found: " + name);
+
+        LocalStringServerSocket ss = (LocalStringServerSocket) allSockets.get(name);       
+        if (ss.isOutEOF())
+            throw new ConnectException("LocalStringServerSocket name is EOF: " + name);
+
+        LocalStringConnection servSidePeer;
+        try
+        {
+            servSidePeer = ss.queueAcceptClient(client);
+        }
+        catch (ConnectException ce)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)